src/main/java/com/celements/mandatory/HTMLWebPreferences.java

```diff
@@ -90,31 +94,19 @@ boolean isSkipCelementsHTMLwebPreferences() {
 
   void checkHTMLwebPreferences() throws XWikiException {
     DocumentReference htmlWebPreferencesRef = getHTMLwebPreferencesRef(getContext().getDatabase());
-    XWikiDocument wikiPrefDoc;
-    if (!getContext().getWiki().exists(htmlWebPreferencesRef, getContext())) {
-      LOGGER.debug("HTMLwebPreferencesDocument is missing that we create it. ["
-          + getContext().getDatabase() + "]");
-      wikiPrefDoc = new CreateDocumentCommand().createDocument(htmlWebPreferencesRef,
-          _SPACE_PREFERENCE_PAGE_TYPE);
-    } else {
-      wikiPrefDoc = getContext().getWiki().getDocument(htmlWebPreferencesRef, getContext());
-      LOGGER.trace("HTMLwebPreferencesDocument already exists. [" + getContext().getDatabase()
-          + "]");
-    }
-    if (wikiPrefDoc != null) {
-      boolean dirty = checkPageType(wikiPrefDoc);
-      dirty |= checkHTMLwebPreferences(wikiPrefDoc);
-      if (dirty) {
+    XWikiDocument wikiPrefDoc = modelAccess.getOrCreateDocument(htmlWebPreferencesRef);
+    boolean dirty = checkPageType(wikiPrefDoc);
+    dirty |= checkHTMLwebPreferences(wikiPrefDoc);
+    if (dirty) {
+      try {
         LOGGER.info("HTMLwebPreferencesDocument updated for [" + getContext().getDatabase() + "].");
-        getContext().getWiki().saveDocument(wikiPrefDoc, "autocreate" + " HTML.WebPreferences.",
-            getContext());
-      } else {
-        LOGGER.debug("HTMLwebPreferencesDocument not saved. Everything uptodate. ["
-            + getContext().getDatabase() + "].");
+        modelAccess.saveDocument(wikiPrefDoc, "autocreate HTML.WebPreferences.");
+      } catch (DocumentSaveException dse) {
+        throw new XWikiException(0, 0, "failed saving", dse);
       }
     } else {
-      LOGGER.trace("skip checkHTMLwebPreferences because wikiPrefDoc is null! ["
-          + getContext().getDatabase() + "]");
+      LOGGER.debug("HTMLwebPreferencesDocument not saved. Everything uptodate. ["
+          + getContext().getDatabase() + "].");
```

Comment:
```suggestion
      LOGGER.debug("HTMLwebPreferencesDocument not saved. Everything uptodate. [{}]",
          getContext().getDatabase());
```
why not using slf4j message formatting?